proto/api.go

```diff
@@ -308,14 +307,139 @@ func (sr *ReverseScanResponse) Verify(req Request) error {
 	return nil
 }
 
+// GetInner returns the Request contained in the `oneof`.
+func (ru BatchRequest_RequestUnion) GetInner() Request {
```

Comment:
> I think it was implemented this way so that ru.Request could enforce that only certain types be assigned. 

I think (but haven't confirmed) that `oneof` allows multiple fields of the same type (like C `union`), and that the per-field types are necessary to disambiguate this case. `onlyone` required all types to be unique and was able to provide a more convenient interface (modulo casting from `interface{}`) as a result.